### PR TITLE
medxpertqa

### LINKED ISSUE #7: MedXpertQA
Dataset: https://huggingface.co/datasets/TsinghuaC3I/MedXpertQA

Only using the text-only subset for now.

### PR DESCRIPTION
**Pull request by srishti-git1110**
- adds a new environment for [MedXpertQA](https://huggingface.co/datasets/TsinghuaC3I/MedXpertQA/viewer/Text/test?views%5B%5D=text_test&row=0)
- closes #7

### CODE REVIEW

**Comment by warner-benjamin** [comment]
Thanks for the PR.

We can't add the test set as the training set, so this should be a test only dataset.

The [MedXpertQA authors have their prompts on GitHub](https://github.com/TsinghuaC3I/MedXpertQA/blob/main/eval/config/prompt_templates.py), let's use theirs instead a custom one so our implementation of MedXpertQA is more comparable to the paper. [OpenCompass also has their prompts](https://github.com/open-compass/opencompass/pull/2002).

Can you also add an option to their "Put your final answer within \\boxed{{}}" prompt to use thinking tags per verifiers `THINK_BOXED_SYSTEM_PROMPT` from `verifiers.utils.data_utils` for reasoning models and `verifiers.ThinkParser` to extract the answers from a reasoning model? There's also `extract_boxed_answer`?

**Comment by warner-benjamin** [comment]
@srishti-git1110  The <think> tags should only be used if `use_think` is passed to the environment, otherwise we want the "Let's think step by step" prompt. That is when we want to include the `ThinkParser`. You can see an example of this in #19.

**Comment by srishti-git1110** [comment]
Thank you @warner-benjamin. Does it look fine now? Happy to revise more as needed. :)
